Source/Core/DolphinQt2/Config/LoggerWidget.cpp

```diff
@@ -0,0 +1,435 @@
+
+  auto* config_types = new QGroupBox(tr("Log Types"));
+  auto* types_layout = new QVBoxLayout;
+  config_types->setLayout(types_layout);
+  m_types_toggle = new QPushButton(tr("Toggle All Log Types"));
+  m_types_list = new QListWidget;
+
+  for (int i = 0; i < LogTypes::NUMBER_OF_LOGS; i++)
+  {
+    QListWidgetItem* widget = new QListWidgetItem(QString::fromStdString(
+        LogManager::GetInstance()->GetFullName(static_cast<LogTypes::LOG_TYPE>(i))));
+    widget->setCheckState(Qt::Unchecked);
+    m_types_list->addItem(widget);
+  }
+
+  config_layout->addWidget(config_verbosity);
+  verbosity_layout->addWidget(m_verbosity_notice);
+  verbosity_layout->addWidget(m_verbosity_error);
+  verbosity_layout->addWidget(m_verbosity_warning);
+  verbosity_layout->addWidget(m_verbosity_info);
+
+  config_layout->addWidget(config_outputs);
+  outputs_layout->addWidget(m_out_file);
+  outputs_layout->addWidget(m_out_console);
+  outputs_layout->addWidget(m_out_window);
+
+  config_layout->addWidget(config_types);
+  types_layout->addWidget(m_types_toggle);
+  types_layout->addWidget(m_types_list);
+}
+
+void LoggerWidget::CreateMainLayout()
+{
+  QWidget* widget = new QWidget;
+  QVBoxLayout* layout = new QVBoxLayout;
+
+  widget->setLayout(layout);
+  layout->addWidget(m_tab_widget);
+
+  setWidget(widget);
+}
+
+void LoggerWidget::ConnectWidgets()
+{
+  // Configuration
+  connect(m_verbosity_notice, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_error, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_warning, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_verbosity_info, &QRadioButton::toggled, this, &LoggerWidget::SaveSettings);
+
+  connect(m_out_file, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_out_console, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_out_window, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+
+  connect(m_types_toggle, &QPushButton::clicked, [this] {
+    m_all_enabled = !m_all_enabled;
+
+    // Don't save every time we change an item
+    m_block_save = true;
+
+    for (int i = 0; i < m_types_list->count(); i++)
+      m_types_list->item(i)->setCheckState(m_all_enabled ? Qt::Checked : Qt::Unchecked);
+
+    m_block_save = false;
+
+    SaveSettings();
+  });
+
+  connect(m_types_list, &QListWidget::itemChanged, this, &LoggerWidget::SaveSettings);
+
+  // Log
+  connect(m_log_clear, &QPushButton::clicked, m_log_text, &QTextEdit::clear);
+  connect(m_log_wrap, &QCheckBox::toggled, this, &LoggerWidget::SaveSettings);
+  connect(m_log_font, static_cast<void (QComboBox::*)(int)>(&QComboBox::currentIndexChanged), this,
+          &LoggerWidget::SaveSettings);
+
+  // Window tracking
+  connect(this, &QDockWidget::topLevelChanged, this, &LoggerWidget::SaveSettings);
+}
+
+void LoggerWidget::LoadSettings()
+{
+  m_ini.Load(File::GetUserPath(F_LOGGERCONFIG_IDX));
+
```

Comment:
I think that'd be alright. I'm not really sure either way on these UI-only configured values.